src/project/internal/projectactionscontroller.cpp

```diff
@@ -175,7 +175,7 @@ Ret ProjectActionsController::openProject(const ProjectFile& file)
 {
     LOGI() << "Try open project: url = " << file.url.toString() << ", displayNameOverride = " << file.displayNameOverride;
 
-    if (file.url.isLocalFile()) {
+    if (file.isNull() || file.url.isLocalFile()) {
```

Comment:
At first glance, I thought there was a typo in your solution (I was expecting this condition: `if (!file.isNull() && file.url.isLocalFile()`) ... Ideally, the implementation should look like the following:

```
if (file.isNull()) {
    path = askUser();
    openProject(path);
} else if (file.url.isLocalFile()) {
    openProject(file)
}
```

But at least it's worth adding a comment

Reply:
Yeah, I agree that's much nicer. Done!